distributed/comm/utils.py

```diff
@@ -1,43 +1,88 @@
 from __future__ import print_function, division, absolute_import
 
+from concurrent.futures import ThreadPoolExecutor
 import logging
 import socket
 
+from tornado import gen
+
 from .. import protocol
-from ..utils import get_ip, get_ipv6
+from ..compatibility import finalize
+from ..sizeof import sizeof
+from ..utils import get_ip, get_ipv6, nbytes
 
 
 logger = logging.getLogger(__name__)
 
 
+# Offload (de)serializing large frames to improve event loop responsiveness
+FRAME_OFFLOAD_THRESHOLD = 10 * 1024 ** 2   # 10 MB
+
+
+class Offloader(object):
+
+    def __init__(self):
+        self.executor = ThreadPoolExecutor(max_workers=1)
+        self._finalizer = finalize(self, self.executor.shutdown)
+
+    def submit(self, fn, *args, **kwargs):
+        return self.executor.submit(fn, *args, **kwargs)
+
+
+_offloader = Offloader()
+
+
+def offload(fn, *args, **kwargs):
+    return _offloader.submit(fn, *args, **kwargs)
```

Comment:
Why the need for the `Offloader` class here?  Would a raw `ThreadPoolExecutor` work just as well?  Is this necessary because of the `finalize` cleanup code?  Does finalize require being part of a class?